dnngior/fasta2model_CLI.py

```diff
@@ -0,0 +1,249 @@
+        # if not os.path.exists(extra_folder):
+        #     print('# Creating gapfill info folder')
+        #     os.makedirs(extra_folder, exist_ok=True)
+        # else:
+        #     print('# WARNING: tmp folder already exists')
+    else:
+        sys.exit('# ERROR: your output_folder should have parents')
+
+def main():
+    args = parse_arguments()
+    create_output_folder(args)
+    logging.basicConfig(filename=os.path.join(args.output_folder, 'gapfill.log'), filemode='a', format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO, datefmt='%Y-%m-%d %H:%M:%S')
+
+    #I am making my life way too complicated
+    gf_data_location = os.path.join(args.output_folder, 'gf_data.tsv')
+    if os.path.isfile(gf_data_location):
+        file_count = 0
+        gf_data_location = os.path.join(args.output_folder, f'gf_data_{file_count}.tsv')
+        while os.path.isfile(gf_data_location):
+            file_count += 1
+            gf_data_location = gf_data_location.replace(str(file_count-1),str(file_count))
+
+    args.gf_data_file = open(gf_data_location, 'w')
+    args.gf_data_file.write('model_id\tn_reactions_base\tn_gf_reactions\tn_total_reactions\tmedium\n')
+
+
+    if args.fasta_folder:
+        list_of_genomes = [i for i in os.listdir(args.fasta_folder) if i.endswith(args.suffix_faa) or i.endswith(args.suffix_faa+'.gz')]
+        len_list_of_genomes = len(list_of_genomes)
+        if len_list_of_genomes == 0:
+            sys.exit(f'# ERROR: no fasta files found in {args.fasta_folder}')
+        logging.info(f'# Building and gapfilling {len_list_of_genomes} models')
+        for i, genome in enumerate(list_of_genomes):
+            args.path_to_fasta = os.path.join(args.fasta_folder, genome)
+            args.model_name = '.'.join(os.path.basename(args.path_to_fasta).split('.')[:-1])
+            logging.info(f'# Building and gapfilling: {genome} ({i+1}/{len_list_of_genomes})')
+            build_gapfilled_model_from_fasta(args)
+        logging.info('#Done')
+
+    elif args.model_folder:
+        list_of_models = [i for i in os.listdir(args.model_folder) if i.endswith(args.suffix_model)]
+        len_list_of_models = len(list_of_models)
+        if len_list_of_models == 0:
+            sys.exit(f'# ERROR: no model files found in {args.model_folder}')
+        logging.info(f'# Gapfilling {len_list_of_models} models')
+        for i, model in enumerate(list_of_models):
+            args.path_to_base_model = os.path.join(args.model_folder, model)
+            args.model_name = '.'.join(os.path.basename(args.path_to_base_model).split('.')[:-1])
+            if args.model_name.startswith('base_'):
+                args.model_name = args.model_name[5:]
+            logging.info(f'# Gapfilling: {model} ({i+1}/{len_list_of_models})')
+            gapfill_model_wrapper(args)
+        print('# Done')
+    else:
+        sys.exit('I dont think this message can show, if it does, you (or more likely me) did something weird')
```

Comment:
Consider editing this message returning something like 
"either a fasta_folder or a model_folder needs to be provided"